pkg/controller/application_spec_sync.go

```diff
@@ -0,0 +1,50 @@
+package controller
+
+import (
+	"fmt"
+
+	"github.com/jackc/pgx/v4/pgxpool"
+	"k8s.io/apimachinery/pkg/api/equality"
+	appsv1 "sigs.k8s.io/application/api/v1beta1"
+	ctrl "sigs.k8s.io/controller-runtime"
+	"sigs.k8s.io/controller-runtime/pkg/client"
+)
+
+func areApplicationsEqual(instance1, instance2 client.Object) bool {
```

Comment:
for the sake of code organization and consistency, I would use the same order as in other spec sync files.
that is - first the add controller, then clean status func, and then are equal func.
see for example here - https://github.com/stolostron/hub-of-hubs-spec-sync/blob/main/pkg/controller/placementbinding_spec_sync.go